File: cpp/39b_1296A_ArrayWithOddSum.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, cnt, temp, no_of_odd;
    std::vector<std::string> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        cnt = 0, no_of_odd = 0;
        for (int j = 0; j < n; j++){
            std::cin>>temp;
            cnt += temp;
            if(temp % 2 != 0){
                no_of_odd++;
            }
        }
        if (no_of_odd == 0){
            ans.push_back("NO");
        }
        else{
            if(cnt % 2 == 0 && no_of_odd == n){
                ans.push_back("NO");
            }
            else{
                ans.push_back("YES");
            }
        }
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/39b_1296A_ArrayWithOddSum.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = input().split()
        cnt, no_of_odd = 0, 0
        for i in arr:
            cnt += int(i)
            if int(i) % 2 != 0:
                no_of_odd += 1
        if no_of_odd == 0:
            ans.append("NO")
        else:
            if cnt % 2 == 0 and no_of_odd == n:
                ans.append("NO")
            else:
                ans.append("YES")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()